06_puppy/lexual/puppy.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+var out io.Writer = os.Stdout
+
+type Puppy struct {
```

Comment:
Please include documentation for all exported types. It is good practice to get in the habit for doing this from now on.